### DISCUSSION TITLE
WHERE on INNER JOIN

### DISCUSSION THREAD

**Discussion by collinalexbell** [prisma/prisma#16305] [Q&A] [answered]
Hi

I want

`SELECT * from customers INNER JOIN orders on customers.id = orders.customer.id WHERE orders.product_type = 'digital' AND customers.subscribed_to_emails = true`

which would only perform INNER JOINs on customers that have subscribed_to_emails = true

In prisma, I would assume something like this would work:

```
prisma.customers.findMany({where: {order: {productType: "digital", subscribedToEmails: true}}}
```

but that generates a WHERE clause with an INNER JOIN in it, which is incredibly inefficient, because it is joining all customers to orders, when I only need customers subscribed to emails to be joined. 

Is there a way to make the SQL query I want with prisma (not raw query)?

**Comment by nurul3101** [comment, marked as answer]
Hello @kuberlog 👋 

Just to confirm, you enabled [logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging) and checked the SQL Query which was generated by invoking the findMany query, correct?

I think this GitHub Issue: [#14688](https://github.com/prisma/prisma/issues/14688) is similar to what you are experiencing. Can you have a look? 

For now, I think the Raw query would be the only way to run the exact query that you have shared.

**Reply by collinalexbell** [reply]
Yep. That is the problem. The generated query is gnarly with inner joins in the WHERE

**Reply by nurul3101** [reply]
The team is aware of this, and it should be tackled when we work on performance improvements. Thanks for the feedback 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
